lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -1451,6 +1467,40 @@ llvm::Expected<CompilerType> SwiftLanguageRuntime::GetChildCompilerTypeAtIndex(
                                    "is out of bounds (" + llvm::Twine(i - 1) +
                                    ")");
   }
+  // Fixed array.
+  if (auto *ati = llvm::dyn_cast<swift::reflection::ArrayTypeInfo>(ti)) {
+    LLDB_LOG(GetLog(LLDBLog::Types), "{0}: ArrayTypeInfo()",
+             type.GetMangledTypeName().GetCString());
+    auto *el_ti = ati->getElementTypeInfo();
+    if (!el_ti)
+      return llvm::createStringError("array without element type info: " +
+                                     type.GetMangledTypeName().GetString());
+    child_name.clear();
+    llvm::raw_string_ostream(child_name) << idx;
+    child_byte_size = el_ti->getSize();
+    child_byte_offset = el_ti->getStride() * idx;
+    if (!ignore_array_bounds &&
+        (int64_t)child_byte_offset > (int64_t)ati->getSize())
+      return llvm::createStringError("array index out of bounds");
+
+    child_bitfield_bit_size = 0;
+    child_bitfield_bit_offset = 0;
+    child_is_base_class = false;
+    child_is_deref_of_parent = false;
+    language_flags = 0;
+
+    swift::Demangle::Demangler dem;
+    swift::Demangle::NodePointer global =
+        dem.demangleSymbol(type.GetMangledTypeName().GetStringRef());
+    using Kind = Node::Kind;
+    auto *dem_array_type = swift_demangle::ChildAtPath(
+        global, {Kind::TypeMangling, Kind::Type, Kind::BuiltinFixedArray});
+    if (!dem_array_type || dem_array_type->getNumChildren() != 2)
+      return llvm::createStringError("Expected fixed array, but found: " +
+                                     type.GetMangledTypeName().GetString());
+    return ts->RemangleAsType(dem, dem_array_type->getChild(1),
+                              ts->GetManglingFlavor());
```

Comment:
```
    auto flavor = SwiftLanguageRuntime::GetManglingFlavor(
            type.GetMangledTypeName().GetStringRef());
    return ts->RemangleAsType(dem, dem_array_type->getChild(1), flavor);

Reply:
> At some point it would be nice to split these functions into smaller functions that deal with the different type info types. GetChildCompilerTypeAtIndex will be almost 400 lines long with this change.

I think the way to go is to factor out common code between GetChildCompilerTypeAtIndex, GetNumChildren, and GetChildMembeWithName.